### DISCUSSION TITLE
Adding highlighting for Ligo, a language with many syntaxes

### DISCUSSION THREAD

**Discussion by pewulfman** [github-linguist/linguist#5733] [Q&A] [answered]
Hello all,
I'm a lead engineer at [LigoLang](https://ligolang.org/), a smart-contract language whose development is financed by the Tezos foundation. My point is that we are not yet another toy language :sweat_smile: . We actually have a growing user database in the blockchain community. And we would like to have the language be recognized by Github. So I assume all we have to do is contribute to the linguist repository, which provides all the documentation.

I am reading it at the moment and I'm foreshadowing a problem that I think is quite specific to Ligo. We have multiple syntaxes, so different file extensions which are connected to different syntax highlighters, but for the same language. 
So I could define separate languages for each syntax that we have, which will solve the issues but then it will be reported as a different language which would be wrong (You can think of our syntaxes are dialects of the same languages... except they are completely different... but compatible inside the same project i.e. you can import a file in jsligo into a file in cameligo since the language is the same)

I will look into the source code but I assume this use-case doesn't exist yet and wasn't planned, does it? 
Do you have some idea if that is possible and how to do it ? (I guess adding the idea of "dialect" as a subcategory of language ?)

**Comment by Nixinova** [comment, marked as answer]
> different file extensions which are connected to different syntax highlighters, but for the same language.

This is done by adding each syntax as a separate languag then grouping all these dialects under a parent language category using the `group` key in the languages.yml file.

Each file extension needs to be used in 200+ repos to be added to linguist, however.

**Reply by pewulfman** [reply]
Thank :) that's perfect

**Comment by pewulfman** [comment]
Thanks for your replies, I have made the pull request : https://github.com/github/linguist/pull/5811
